### PR TITLE
Deprecate `train_test_split_edges` + Modifications to `RandomLinkSplit`

### PR DESCRIPTION
**Pull request by rusty1s**
This PR applies the following changes:
* It deprecates `train_test_split_edges` in favor of our new `RandomLinkSplit` transform
* It modifies our link prediction examples accordingly
* It adds a new transform called `ToDevice`, which takes care of converting graphs to the GPU
* It adds an additional argument to `RandomLinkSplit`, which allows to separate positive and negative labels by saving them in different graph storage attributes. This is useful when combining `RandomLinkSplit` with our `Autoencoder` model.

### CODE REVIEW

File: examples/argva_node_clustering.py [hunk 1]
```diff
@@ -1,88 +1,86 @@
...
-encoder_optimizer = torch.optim.Adam(encoder.parameters(), lr=0.005)
 
 
 def train():
     model.train()
     encoder_optimizer.zero_grad()
-    z = model.encode(data.x, data.train_pos_edge_index)
+    z = model.encode(train_data.x, train_data.edge_index)
 
     for i in range(5):
```
Comment (RexYing) [inline examples/argva_node_clustering.py:66]: It might be a bit unrelated to this CL. Is it possible to specify 5 as a constant / argparse arguments?  e.g. max_iter or sth?

File: examples/argva_node_clustering.py [hunk 2]
```diff
@@ -1,88 +1,86 @@
...
-data = train_test_split_edges(data)
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+transform = T.Compose([
+    T.ToDevice(device),
+    T.RandomLinkSplit(num_val=0.05, num_test=0.1, is_undirected=True,
+                      split_labels=True, add_negative_train_samples=False),
+])
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Planetoid')
+dataset = Planetoid(path, name='Cora', transform=transform)
+train_data, val_data, test_data = dataset[0]
```
Comment (RexYing) [inline examples/argva_node_clustering.py:23]: maybe add a comment here to make it more friendly to users that are not familiar? 
e.g.
\# After applying T.RandomLinkSplit, the dataset is transformed from a list of Data object to a 
\# list to tuples (train_data, val_data, test_data), with each element being a Data object corresponding to the split.

File: examples/seal_link_pred.py [hunk 3]
```diff
@@ -197,6 +189,7 @@ def forward(self, x, edge_index, batch):
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 model = DGCNN(hidden_channels=32, num_layers=3).to(device)
 optimizer = torch.optim.Adam(params=model.parameters(), lr=0.0001)
```
Comment (RexYing) [inline examples/seal_link_pred.py:191]: if optimizer and criterion is train function specific maybe we can put it in train()

File: torch_geometric/transforms/random_link_split.py [hunk 4]
```diff
@@ -45,6 +47,10 @@ class RandomLinkSplit(BaseTransform):
             After negative sampling, label :obj:`0` represents negative edges,
             and labels :obj:`1` to :obj:`num_classes` represent the labels of
             positive edges. (default: :obj:`"edge_label"`)
+        split_labels (bool, optional): If set to :obj:`True`, will split
```
Comment (RexYing) [inline torch_geometric/transforms/random_link_split.py:50]: I think it's ok to have this functionality. Just curious when is it needed? Is it for compatibility purpose?

**Review by RexYing — APPROVED** [review]
thanks for the example updates! I think it's good to merge. i only listed a few minor comments.

File: torch_geometric/transforms/random_link_split.py (hunk 4 above)
Comment (rusty1s) [reply torch_geometric/transforms/random_link_split.py:50]: Yeah, it's for backward compatibility with the `GAE` class, which takes in `pos_edge_index` and `neg_edge_index` for computing the loss.

**Comment by rusty1s** [comment]
Addressed the comments. Thanks for having a look!
